### PR TITLE
Reverted Changes to Git Config and Authentication

### PR DESCRIPTION
**Pull request by thboop**
The updates were breaking user's builds #60, #62.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,4 +15,5 @@ jobs:
     steps:
     - uses: actions/checkout@master
     - uses: ./
-    - run: git ls-remote --tags origin
+      with:
```
Comment (thboop) [inline .github/workflows/test.yml]: We had a flow to test the v1.2 feature, so we need to update our tests on the revert

File: README.md [hunk 2]
```diff
@@ -25,39 +25,43 @@ steps:
...
+> - Private submodules must be configured via `https` not `ssh`.
+> - `${{ github.token }}` only has permission to the workflow triggering repository. If the repository contains any submodules that come from private repositories, you will need to add your [PAT](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line) as secret and use the secret in `with.token` to make the `checkout` action work.
 
 For more details, see [Contexts and expression syntax for GitHub Actions](https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions) and [Creating and using secrets (encrypted variables)](https://help.github.com/en/articles/virtual-environments-for-github-actions#creating-and-using-secrets-encrypted-variables)
 
 # Changelog
 
-## v1.1.0 (unreleased)
+## v1.2.0
+- Reverted persisting auth token changes to fix a bug with custom auth flows
```
Comment (bryanmacfarlane) [inline README.md]: I would say "Reverted the breaking behavior change in v1.1.0 ..." so it's clear

File: README.md [hunk 3]
```diff
@@ -25,39 +25,43 @@ steps:
...
+
+## v1.1.0
```
Comment (bryanmacfarlane) [inline README.md]: Update this with a note to do not use this behavior.  The (breaking) functionality intended with this release will be in the 2.0 release.

File: README.md [hunk 4]
```diff
@@ -25,39 +25,43 @@ steps:
...
-> - `${{ github.token }}` only has permission to the workflow triggering repository. If the repository contains any submodules that comes from private repository, you will have to add your PAT as secret and use the secret in `with.token` to make `checkout` action work.
+> - Private submodules must be configured via `https` not `ssh`.
+> - `${{ github.token }}` only has permission to the workflow triggering repository. If the repository contains any submodules that come from private repositories, you will need to add your [PAT](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line) as secret and use the secret in `with.token` to make the `checkout` action work.
 
 For more details, see [Contexts and expression syntax for GitHub Actions](https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions) and [Creating and using secrets (encrypted variables)](https://help.github.com/en/articles/virtual-environments-for-github-actions#creating-and-using-secrets-encrypted-variables)
 
 # Changelog
```
Comment (bryanmacfarlane) [inline README.md]: Let's move this into a CHANGELOG.md file.

File: README.md [hunk 5]
```diff
@@ -25,39 +25,43 @@ steps:
 - run: npm test
 ```
```
Comment (bryanmacfarlane) [inline README.md]: Add a note somewhere in this readme to not reference `master` and link to the versioning doc in the toolkit

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# Changelog
+
+## Unreleased Changes
+- N/A
+
+## v1.2.0
+- Reverted the breaking behavior change in v1.1.0 that broke custom authentication flows
+
+## v1.1.0 (Not reccomended for use, this functionality will be ported to the 2.0 update)
```
Comment (ericsciple) [inline CHANGELOG.md]: why mention 1.1.0 at all? why burden users with groking this knowledge that they dont need

in the same spirit, why v1.2.0?

at most should we just saw `## v1.1.0 (unreleased)`

File: README.md [hunk 7]
```diff
@@ -25,40 +25,39 @@ steps:
...
+> - `${{ github.token }}` is scoped to the current repository, so if you want to checkout another repository that is private you will need to provide your own [PAT](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line).
 
-Checkout private submodules:
+## Checkout private submodules
 ```yaml
 - uses: actions/checkout@v1
   with:
-    submodules: recursive
+    submodules: true // 'recursive' 'true' or 'false'
```
Comment (ericsciple) [inline README.md]: `#`

**Comment by Imebeez** [comment]
[]()

**Pull request merged**